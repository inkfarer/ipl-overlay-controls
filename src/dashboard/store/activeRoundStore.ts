import { ActiveRound, SwapColorsInternally } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { GameWinner } from 'types/enums/gameWinner';
import { SetActiveColorRequest, SetWinnerRequest, SwapRoundColorRequest } from 'types/messages/activeRound';
import { SetRoundRequest } from 'types/messages/rounds';
import { ActiveRoundGame } from 'types/activeRoundGame';

const activeRound = nodecg.Replicant<ActiveRound>('activeRound');
const swapColorsInternally = nodecg.Replicant<SwapColorsInternally>('swapColorsInternally');

export const activeRoundReps = [ activeRound, swapColorsInternally ];

export interface ActiveRoundStore {
    activeRound: ActiveRound;
    swapColorsInternally: SwapColorsInternally;
}

export const activeRoundStore = createStore<ActiveRoundStore>({
    state: {
        activeRound: null,
        swapColorsInternally: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        setWinner(store, { winner }: { winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner });
        },
        removeWinner(): void {
            nodecg.sendMessage('removeWinner');
        },
        setActiveColor(store, request: SetActiveColorRequest): void {
            nodecg.sendMessage('setActiveColor', request);
        },
        swapColors(): void {
            swapColorsInternally.value = !swapColorsInternally.value;
        },
        setWinnerForIndex(store, { index, winner }: { index: number, winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner, roundIndex: index } as SetWinnerRequest);
        },
        resetActiveRound(): void {
            nodecg.sendMessage('resetActiveRound');
        },
        setActiveRound(store, request: SetRoundRequest): void {
            nodecg.sendMessage('setActiveRound', request);
        },
        swapRoundColor(store, request: SwapRoundColorRequest): void {
            nodecg.sendMessage('swapRoundColor', request);
        },
        updateActiveGames(store, games: ActiveRoundGame[]): void {
            nodecg.sendMessage('updateActiveGames', { games });
        },
        switchToNextColor(): void {
            nodecg.sendMessage('switchToNextColor');
        },
        switchToPreviousColor(): void {
            nodecg.sendMessage('switchToPreviousColor');
        },
        async getNextAndPreviousColors(): Promise<void> {
            return nodecg.sendMessage('getNextAndPreviousColors');
        }
    }
});

export const activeRoundStoreKey: InjectionKey<Store<ActiveRoundStore>> = Symbol();

export function useActiveRoundStore(): Store<ActiveRoundStore> {
    return useStore(activeRoundStoreKey);
}
