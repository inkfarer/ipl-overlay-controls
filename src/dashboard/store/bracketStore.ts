import { Bracket } from '@tourneyview/common';
import { defineStore } from 'pinia';

const bracketData = nodecg.Replicant<Bracket | undefined>('bracketData');

export const bracketReps = [ bracketData ];

export interface BracketStore {
    bracketData: Bracket | undefined;
}

export const useBracketStore = defineStore('brackets', {
    state: () => ({
        bracketData: undefined
    } as BracketStore),
    actions: {
        resetBracketData() {
            bracketData.value = undefined;
        }
    }
});
