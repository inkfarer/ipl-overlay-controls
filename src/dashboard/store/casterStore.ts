import { NodeCGBrowser } from 'nodecg/browser';
import { Caster, Casters, RadiaSettings } from 'schemas';
import { generateId } from '../../helpers/generateId';
import { defineStore } from 'pinia';

const casters = nodecg.Replicant<Casters>('casters');
const radiaSettings = nodecg.Replicant<RadiaSettings>('radiaSettings');

export const castersReps = [ casters, radiaSettings ];

export interface CasterStore {
    casters: Casters;
    uncommittedCasters: Casters;
    radiaSettings: RadiaSettings;
}

const defaultCaster = {
    name: 'New Caster',
    twitter: '',
    pronouns: '?/?'
};

export const useCasterStore = defineStore('casters', {
    state: () => ({
        casters: {},
        uncommittedCasters: {},
        radiaSettings: {
            enabled: null,
            guildID: null,
            updateOnImport: null
        }
    } as CasterStore),
    actions: {
        updateCaster({ id, newValue }: { id: string, newValue: Caster }): void {
            casters.value[id] = newValue;
        },
        removeUncommittedCaster(id: string): void {
            delete this.uncommittedCasters[id];
        },
        addUncommittedCaster({ id, caster }: { id: string, caster?: Caster }): void {
            this.uncommittedCasters[id] = caster ?? defaultCaster;
        },
        addDefaultCaster(): string {
            const id = generateId();
            this.addUncommittedCaster({ id });
            return id;
        },
        async saveUncommittedCaster({ id, caster }: { id: string, caster: Caster }): Promise<string> {
            const newId = await nodecg.sendMessage('saveCaster', caster);
            this.removeUncommittedCaster(id);
            return newId;
        },
        async loadCastersFromVc(): Promise<void> {
            const result = await nodecg.sendMessage('getLiveCommentators');
            if (result.extra && result.extra.length > 0) {
                for (let i = 0; i < result.extra.length; i++) {
                    const extraCaster = result.extra[i];
                    const id = extraCaster.discord_user_id;
                    delete extraCaster.discord_user_id;
                    this.addUncommittedCaster({ id, caster: extraCaster });
                }
            }
        },
        showCasters() {
            nodecg.sendMessage('mainShowCasters');
        },
        async removeCaster(id: string): Promise<void> {
            return nodecg.sendMessage('removeCaster', { id });
        }
    }
});
