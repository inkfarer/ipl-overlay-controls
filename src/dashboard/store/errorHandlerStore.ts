import { App } from 'vue';
import { generateId } from '../../helpers/generateId';
import { defineStore } from 'pinia';

export const useErrorHandlerStore = defineStore('errorHandler', {
    state: () => ({
        recentErrors: {}
    }),
    actions: {
        removeRecentError({ key }: { key: string }): void {
            delete this.recentErrors[key];
        },
        handleError({ err, info }: { err: unknown, info: string }): void {
            console.error(`Got error from '${info}': \n`, err);

            if (Object.keys(this.recentErrors).length >= 2) return;

            const id = generateId();
            this.recentErrors[id] = err;
            window.setTimeout(() => {
                delete this.recentErrors[id];
            }, 25000);
        }
    }
});

export function setUpErrorHandler(app: App<unknown>): void {
    const store = useErrorHandlerStore();
    app.config.errorHandler = (err, vm, info) => {
        store.handleError({ err, info });
    };
}
