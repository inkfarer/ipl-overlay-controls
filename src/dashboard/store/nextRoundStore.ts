import { NextRound } from 'schemas';
import { SetNextRoundRequest } from 'types/messages/rounds';
import { BeginNextMatchRequest } from 'types/messages/activeRound';
import { defineStore } from 'pinia';

const nextRound = nodecg.Replicant<NextRound>('nextRound');

export const nextRoundReps = [ nextRound ];

export interface NextRoundStore {
    nextRound: NextRound;
}

export const useNextRoundStore = defineStore('nextRound', {
    state: () => ({
        nextRound: null
    } as NextRoundStore),
    actions: {
        setShowOnStream(newValue: boolean) {
            nextRound.value.showOnStream = newValue;
        },
        beginNextMatch(request: BeginNextMatchRequest): void {
            nodecg.sendMessage('beginNextMatch', request);
        },
        setNextRound(request: SetNextRoundRequest): void {
            nodecg.sendMessage('setNextRound', request);
        }
    }
});
