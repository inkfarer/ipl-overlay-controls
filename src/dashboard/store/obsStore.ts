import { NodeCGBrowser } from 'nodecg/browser';
import { ObsCredentials, ObsData } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { SetObsDataRequest } from 'types/messages/obs';

const obsData = nodecg.Replicant<ObsData>('obsData');
const obsCredentials = nodecg.Replicant<ObsCredentials>('obsCredentials');

export const obsReps = [obsData, obsCredentials];

export interface ObsStore {
    obsData: ObsData
    obsCredentials: ObsCredentials
}

export const obsStore = createStore<ObsStore>({
    state: {
        obsData: null,
        obsCredentials: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        async connect(store, { address, password }: { address: string, password?: string }): Promise<void> {
            return nodecg.sendMessage('connectToObs', { address, password });
        },
        async setData(store, data: SetObsDataRequest): Promise<void> {
            return nodecg.sendMessage('setObsData', data);
        },
        async startGame(): Promise<void> {
            return nodecg.sendMessage('startGame');
        },
        async endGame(): Promise<void> {
            return nodecg.sendMessage('endGame');
        },
        setEnabled(store, enabled: boolean): void {
            nodecg.sendMessage('setObsSocketEnabled', enabled);
        }
    }
});

export const obsStoreKey: InjectionKey<Store<ObsStore>> = Symbol();

export function useObsStore(): Store<ObsStore> {
    return useStore(obsStoreKey);
}
