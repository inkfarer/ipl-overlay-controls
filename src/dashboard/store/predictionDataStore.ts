import { NodeCGBrowser } from 'nodecg/browser';
import { Prediction, PredictionStore } from 'schemas';
import { PredictionStatus } from 'types/enums/predictionStatus';
import { defineStore } from 'pinia';

const predictionStore = nodecg.Replicant<PredictionStore>('predictionStore');

export const predictionReps = [ predictionStore ];

export interface PredictionDataStore {
    predictionStore: PredictionStore;
}

export const usePredictionDataStore = defineStore('prediction', {
    state: () => ({
        predictionStore: null,
    } as PredictionDataStore),
    actions: {
        async lockPrediction() {
            return patchPrediction(
                this.predictionStore.currentPrediction,
                PredictionStatus.LOCKED,
                [ PredictionStatus.ACTIVE ]);
        },
        async cancelPrediction() {
            return patchPrediction(
                this.predictionStore.currentPrediction,
                PredictionStatus.CANCELED,
                [ PredictionStatus.ACTIVE, PredictionStatus.LOCKED ]);
        },
        async resolvePrediction({ winningOutcomeIndex }: { winningOutcomeIndex: number }) {
            if (winningOutcomeIndex > 1 || winningOutcomeIndex < 0) {
                throw new Error(`Cannot resolve prediction with outcome index ${winningOutcomeIndex}.`);
            } else if (!this.predictionStore.currentPrediction?.id) {
                throw new Error('No prediction to resolve.');
            } else if (this.predictionStore.currentPrediction.status !== PredictionStatus.LOCKED ) {
                throw new Error('Can only resolve a locked prediction.');
            }

            return nodecg.sendMessage('patchPrediction', {
                id: this.predictionStore.currentPrediction.id,
                status: PredictionStatus.RESOLVED,
                winning_outcome_id: this.predictionStore.currentPrediction.outcomes[winningOutcomeIndex].id
            });
        },
        async createPrediction(
            { title, teamAName, teamBName, duration }:
                { title: string, teamAName: string, teamBName: string, duration: number }
        ) {
            if (
                [PredictionStatus.ACTIVE, PredictionStatus.LOCKED]
                    .includes(this.predictionStore.currentPrediction?.status as PredictionStatus)
            ) {
                throw new Error('An unresolved prediction already exists.');
            }

            return nodecg.sendMessage('postPrediction', {
                title,
                outcomes: [
                    { title: teamAName },
                    { title: teamBName }
                ],
                prediction_window: duration
            });
        },
        async reconnect() {
            return nodecg.sendMessage('reconnectToRadiaSocket');
        }
    }
});

async function patchPrediction(
    currentPrediction: Prediction,
    status: PredictionStatus,
    allowedForStatuses: PredictionStatus[]
): Promise<void> {
    if (!currentPrediction || !allowedForStatuses.includes(currentPrediction?.status as PredictionStatus)) {
        throw new Error(`Cannot set prediction status at this time. Status must match one of [${allowedForStatuses.join(', ')}]`);
    }

    return nodecg.sendMessage('patchPrediction', {
        id: currentPrediction.id,
        status: status,
    });
}
