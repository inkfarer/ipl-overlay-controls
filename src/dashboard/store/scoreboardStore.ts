import { NodeCGBrowser } from 'nodecg/browser';
import { ScoreboardData } from 'schemas';
import { defineStore } from 'pinia';

const scoreboardData = nodecg.Replicant<ScoreboardData>('scoreboardData');

export const scoreboardReps = [scoreboardData];

export interface ScoreboardStore {
    scoreboardData: ScoreboardData
}

export const useScoreboardStore = defineStore('scoreboard', {
    state: () => ({
        scoreboardData: {
            flavorText: '',
            isVisible: null
        }
    } as ScoreboardStore),
    actions: {
        setFlavorText(flavorText: string): void {
            scoreboardData.value.flavorText = flavorText;
        },
        setScoreboardVisible(isVisible: boolean): void {
            scoreboardData.value.isVisible = isVisible;
        }
    }
});
