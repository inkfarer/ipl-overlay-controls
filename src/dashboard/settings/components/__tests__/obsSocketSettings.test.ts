import ObsSocketSettings from '../obsSocketSettings.vue';
import { ObsStatus } from 'types/enums/ObsStatus';
import { config, mount } from '@vue/test-utils';
import { createTestingPinia, TestingPinia } from '@pinia/testing';
import { useObsStore } from '../../../store/obsStore';
import { IplButton, IplInput } from '@iplsplatoon/vue-components';

describe('ObsSocketSettings', () => {
    let pinia: TestingPinia;

    config.global.stubs = {
        IplInput: true,
        IplButton: true,
        FontAwesomeIcon: true
    };

    beforeEach(() => {
        pinia = createTestingPinia();

        const obsStore = useObsStore();
        obsStore.obsState = { status: ObsStatus.CONNECTED, enabled: true };
        obsStore.obsCredentials = { address: 'localhost:4444', password: 'pwd' };
    });

    it.each(Object.values(ObsStatus))('matches snapshot when status is %s', status => {
        const store = useObsStore();
        store.obsState.status = status;
        const wrapper = mount(ObsSocketSettings, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot when obs socket is disabled', () => {
        const store = useObsStore();
        store.obsState.enabled = false;
        const wrapper = mount(ObsSocketSettings, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('disables connecting if data is invalid', async () => {
        const wrapper = mount(ObsSocketSettings, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent<typeof IplInput>('[name="socketUrl"]').vm.$emit('update:modelValue', '');
        await wrapper.vm.$nextTick();

        expect(wrapper.getComponent('[data-test="socket-connect-button"]').attributes().disabled).toEqual('true');
    });

    it('changes connect button color if data is changed', async () => {
        const wrapper = mount(ObsSocketSettings, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent<typeof IplInput>('[name="socketUrl"]').vm.$emit('update:modelValue', 'https://new-socket-url');
        await wrapper.vm.$nextTick();

        expect(wrapper.getComponent('[data-test="socket-connect-button"]').attributes().color).toEqual('red');
    });

    it('handles connecting to socket', async () => {
        const store = useObsStore();
        store.connect = jest.fn();
        const wrapper = mount(ObsSocketSettings, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent<typeof IplInput>('[name="socketUrl"]').vm.$emit('update:modelValue', 'https://new-socket-url');
        await wrapper.vm.$nextTick();
        wrapper.getComponent<typeof IplButton>('[data-test="socket-connect-button"]').vm.$emit('click');

        expect(store.connect).toHaveBeenCalledWith({ address: 'https://new-socket-url', password: 'pwd' });
    });

    it('resets data on connect button right click', async () => {
        const wrapper = mount(ObsSocketSettings, {
            global: {
                plugins: [pinia]
            }
        });
        const event = new Event(null);
        jest.spyOn(event, 'preventDefault');

        wrapper.getComponent<typeof IplInput>('[name="socketUrl"]').vm.$emit('update:modelValue', 'https://new-socket-url');
        wrapper.getComponent<typeof IplButton>('[data-test="socket-connect-button"]').vm.$emit('rightClick', event);
        await wrapper.vm.$nextTick();

        expect(wrapper.getComponent('[name="socketUrl"]').attributes().modelvalue).toEqual('localhost:4444');
        expect(event.preventDefault).toHaveBeenCalled();
    });
});
