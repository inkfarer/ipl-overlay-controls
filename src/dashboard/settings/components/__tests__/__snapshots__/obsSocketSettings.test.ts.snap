// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObsSocketSettings matches snapshot when obs socket is disabled 1`] = `
"<div>
  <div class=\\"ipl-space color-dark\\" data-v-11cc5f1c=\\"\\">
    <div class=\\"title\\">OBS Socket</div>
    <div class=\\"ipl-toggle__container layout horizontal center-horizontal\\" data-v-1d29c694=\\"\\">
      <div class=\\"ipl-toggle__text\\" data-v-1d29c694=\\"\\"><span class=\\"true-option\\" data-v-1d29c694=\\"\\">Enable</span><span class=\\"false-option selected\\" data-v-1d29c694=\\"\\">Disable</span></div>
      <div class=\\"ipl-toggle__button\\" data-v-1d29c694=\\"\\"></div>
    </div>
  </div>
  <div class=\\"ipl-message__wrapper layout horizontal center-vertical ipl-message__type-info m-t-8\\" data-v-5541e800=\\"\\"><svg class=\\"svg-inline--fa fa-info-circle fa-w-16 icon\\" aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"info-circle\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\" data-v-5541e800=\\"\\">
      <path class=\\"\\" fill=\\"currentColor\\" d=\\"M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z\\"></path>
    </svg>
    <div class=\\"message-content\\" data-v-5541e800=\\"\\">
      <!----> OBS websocket is disabled.
    </div>
  </div>
</div>"
`;

exports[`ObsSocketSettings matches snapshot when status is CONNECTED 1`] = `
"<div>
  <div class=\\"ipl-space color-dark\\" data-v-11cc5f1c=\\"\\">
    <div class=\\"title\\">OBS Socket</div>
    <div class=\\"ipl-toggle__container layout horizontal center-horizontal\\" data-v-1d29c694=\\"\\">
      <div class=\\"ipl-toggle__text\\" data-v-1d29c694=\\"\\"><span class=\\"true-option selected\\" data-v-1d29c694=\\"\\">Enable</span><span class=\\"false-option\\" data-v-1d29c694=\\"\\">Disable</span></div>
      <div class=\\"ipl-toggle__button is-true\\" data-v-1d29c694=\\"\\"></div>
    </div>
  </div>
  <div class=\\"ipl-space color-dark m-t-8\\" data-v-11cc5f1c=\\"\\">
    <ipl-input-stub name=\\"socketUrl\\" label=\\"Socket address\\" modelvalue=\\"localhost:4444\\" type=\\"text\\" centered=\\"false\\" disabled=\\"false\\"></ipl-input-stub>
    <ipl-input-stub name=\\"password\\" label=\\"Password (Optional)\\" modelvalue=\\"pwd\\" type=\\"password\\" centered=\\"false\\" disabled=\\"false\\" class=\\"m-t-4\\"></ipl-input-stub>
    <ipl-button-stub label=\\"Connect\\" color=\\"blue\\" disabled=\\"false\\" async=\\"true\\" progressmessage=\\"Connecting...\\" successmessage=\\"Connected!\\" requiresconfirmation=\\"false\\" shortconfirmationmessage=\\"false\\" small=\\"false\\" disableonsuccess=\\"false\\" href=\\"\\" class=\\"m-t-8\\" data-test=\\"socket-connect-button\\" title=\\"Right-click to undo changes.\\"></ipl-button-stub>
    <div class=\\"ipl-space color-dark text-center m-t-8 text-semibold rounded-inner obs-status_CONNECTED\\" data-v-11cc5f1c=\\"\\">Connected</div>
  </div>
</div>"
`;

exports[`ObsSocketSettings matches snapshot when status is CONNECTING 1`] = `
"<div>
  <div class=\\"ipl-space color-dark\\" data-v-11cc5f1c=\\"\\">
    <div class=\\"title\\">OBS Socket</div>
    <div class=\\"ipl-toggle__container layout horizontal center-horizontal\\" data-v-1d29c694=\\"\\">
      <div class=\\"ipl-toggle__text\\" data-v-1d29c694=\\"\\"><span class=\\"true-option selected\\" data-v-1d29c694=\\"\\">Enable</span><span class=\\"false-option\\" data-v-1d29c694=\\"\\">Disable</span></div>
      <div class=\\"ipl-toggle__button is-true\\" data-v-1d29c694=\\"\\"></div>
    </div>
  </div>
  <div class=\\"ipl-space color-dark m-t-8\\" data-v-11cc5f1c=\\"\\">
    <ipl-input-stub name=\\"socketUrl\\" label=\\"Socket address\\" modelvalue=\\"localhost:4444\\" type=\\"text\\" centered=\\"false\\" disabled=\\"false\\"></ipl-input-stub>
    <ipl-input-stub name=\\"password\\" label=\\"Password (Optional)\\" modelvalue=\\"pwd\\" type=\\"password\\" centered=\\"false\\" disabled=\\"false\\" class=\\"m-t-4\\"></ipl-input-stub>
    <ipl-button-stub label=\\"Connect\\" color=\\"blue\\" disabled=\\"false\\" async=\\"true\\" progressmessage=\\"Connecting...\\" successmessage=\\"Connected!\\" requiresconfirmation=\\"false\\" shortconfirmationmessage=\\"false\\" small=\\"false\\" disableonsuccess=\\"false\\" href=\\"\\" class=\\"m-t-8\\" data-test=\\"socket-connect-button\\" title=\\"Right-click to undo changes.\\"></ipl-button-stub>
    <div class=\\"ipl-space color-dark text-center m-t-8 text-semibold rounded-inner obs-status_CONNECTING\\" data-v-11cc5f1c=\\"\\">Connecting</div>
  </div>
</div>"
`;

exports[`ObsSocketSettings matches snapshot when status is NOT_CONNECTED 1`] = `
"<div>
  <div class=\\"ipl-space color-dark\\" data-v-11cc5f1c=\\"\\">
    <div class=\\"title\\">OBS Socket</div>
    <div class=\\"ipl-toggle__container layout horizontal center-horizontal\\" data-v-1d29c694=\\"\\">
      <div class=\\"ipl-toggle__text\\" data-v-1d29c694=\\"\\"><span class=\\"true-option selected\\" data-v-1d29c694=\\"\\">Enable</span><span class=\\"false-option\\" data-v-1d29c694=\\"\\">Disable</span></div>
      <div class=\\"ipl-toggle__button is-true\\" data-v-1d29c694=\\"\\"></div>
    </div>
  </div>
  <div class=\\"ipl-space color-dark m-t-8\\" data-v-11cc5f1c=\\"\\">
    <ipl-input-stub name=\\"socketUrl\\" label=\\"Socket address\\" modelvalue=\\"localhost:4444\\" type=\\"text\\" centered=\\"false\\" disabled=\\"false\\"></ipl-input-stub>
    <ipl-input-stub name=\\"password\\" label=\\"Password (Optional)\\" modelvalue=\\"pwd\\" type=\\"password\\" centered=\\"false\\" disabled=\\"false\\" class=\\"m-t-4\\"></ipl-input-stub>
    <ipl-button-stub label=\\"Connect\\" color=\\"blue\\" disabled=\\"false\\" async=\\"true\\" progressmessage=\\"Connecting...\\" successmessage=\\"Connected!\\" requiresconfirmation=\\"false\\" shortconfirmationmessage=\\"false\\" small=\\"false\\" disableonsuccess=\\"false\\" href=\\"\\" class=\\"m-t-8\\" data-test=\\"socket-connect-button\\" title=\\"Right-click to undo changes.\\"></ipl-button-stub>
    <div class=\\"ipl-space color-dark text-center m-t-8 text-semibold rounded-inner obs-status_NOT_CONNECTED\\" data-v-11cc5f1c=\\"\\">Not connected</div>
  </div>
</div>"
`;
